### PR TITLE
Port sample to Akka 2.6

### PR DESCRIPTION
**Pull request by chbatey**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Prerequisites:
 
 Open a console and run the following command to apply this template:
  ```
-sbt -Dsbt.version=0.13.15 new akka/akka-quickstart-java.g8
+sbt -Dsbt.version=1.2.8 new akka/akka-quickstart-java.g8
```
Comment (patriknw) [inline README.md]: does it work without this `sbt.version` flag?

File: docs/src/main/paradox/define-actors.md [hunk 2]
```diff
@@ -1,44 +1,77 @@
...
 When defining Actors and their messages, keep these recommendations in mind:
 
-* Since messages are the Actor's public API, _it is a good practice to define messages with good names and rich semantic and domain specific meaning_, even if they just wrap your data type. This will make it easier to use, understand and debug actor-based systems.
+* Since messages are the Actor's public API, _it is a good practice to define messages with good names and rich semantic and domain specific meaning_, even if they just wrap your data type. This will make it easier to use, understand and debug actor-based systems. 
 
-* _Messages should be immutable_, since they are shared between different threads. 
+* _Messages should be immutable_, since they are shared between different threads.
 
-* _It is a good practice to put an actor's associated messages as static classes in the class of the Actor_. This makes it easier to understand what type of messages the actor expects and handles. 
+* It is a good practice to put an actor's associated messages in its object. This makes it easier to understand what type of messages the actor expects and handles.
```
Comment (patriknw) [inline docs/src/main/paradox/define-actors.md]: its object is from Scala, still something like "as static classes in the class of the Actor"

File: docs/src/main/paradox/define-actors.md [hunk 3]
```diff
@@ -1,44 +1,77 @@
...
+* Since messages are the Actor's public API, _it is a good practice to define messages with good names and rich semantic and domain specific meaning_, even if they just wrap your data type. This will make it easier to use, understand and debug actor-based systems. 
 
-* _Messages should be immutable_, since they are shared between different threads. 
+* _Messages should be immutable_, since they are shared between different threads.
 
-* _It is a good practice to put an actor's associated messages as static classes in the class of the Actor_. This makes it easier to understand what type of messages the actor expects and handles. 
+* It is a good practice to put an actor's associated messages in its object. This makes it easier to understand what type of messages the actor expects and handles. 
 
-* _It is also a common pattern to use a static `props` method in the class of the Actor that describes how to construct the Actor._ 
+* It is a good practice obtain an actor's initial behavior in the object's apply method
```
Comment (patriknw) [inline docs/src/main/paradox/define-actors.md]: with a static factory method

File: docs/src/main/paradox/define-actors.md [hunk 4]
```diff
@@ -1,44 +1,77 @@
...
 
-Lets see how the Actor implementations for `Greeter` and `Printer` demonstrate these best practices. 
+Lets see how the objects and Behavior implementations for `Greeter`, `GreeterBot` and `GreeterMain` demonstrate these best practices.
```
Comment (patriknw) [inline docs/src/main/paradox/define-actors.md]: objects?

File: docs/src/main/paradox/define-actors.md [hunk 5]
```diff
@@ -1,44 +1,77 @@
...
-* The `greeting` variable contains the Actor's state and is set to `""` by default.
-* The static `props` method creates and returns a `Props` instance. `Props` is a configuration class to specify options for the creation of actors, think of it as an immutable and thus freely shareable recipe for creating an actor that can include associated deployment information. This example simply passes the parameters that the Actor requires when being constructed. We will see the `props` method in action later in this tutorial.
- 
-### Printer Actor
- 
-The `Printer` implementation is very simple:
+The behavior of the Actor is defined as the `Greeter` with the help
+of the `receive` behavior factory. Processing the next message then results
```
Comment (patriknw) [inline docs/src/main/paradox/define-actors.md]: AbstractBehavior?

File: docs/src/main/paradox/define-actors.md [hunk 6]
```diff
@@ -1,44 +1,77 @@
...
+in a new behavior that can potentially be different from this one. State is
+updated by returning a new behavior that holds the new immutable state. In this
```
Comment (patriknw) [inline docs/src/main/paradox/define-actors.md]: "updated by returning a new behavior" - mutable

File: docs/src/main/paradox/define-actors.md [hunk 7]
```diff
@@ -1,44 +1,77 @@
...
+case we don't need to update any state, so we return @scala[`same`]@java[`this`], which means
```
Comment (patriknw) [inline docs/src/main/paradox/define-actors.md]: scala/java

File: README.md (hunk 1 above)
Comment (chbatey) [reply README.md]: Yep, we can remove, it deals with the user having an old version of sbt on their path

**Pull request merged**